tests/providers/apache/spark/hooks/test_spark_submit.py

```diff
@@ -748,3 +750,64 @@ def test_k8s_process_on_kill(self, mock_popen, mock_client_method):
         client.delete_namespaced_pod.assert_called_once_with(
             'spark-pi-edf2ace37be7353a958b38733a12f8e6-driver',
             'mynamespace', **kwargs)
+
+
+@pytest.mark.parametrize(
+    ("command", "expected"),
+    (
+        (
+            ("spark-submit", "foo", "--bar", "baz", "--password='secret'"),
+            "spark-submit foo --bar baz --password='******'",
+        ),
+        (
+            ("spark-submit", "foo", "--bar", "baz", '--password="secret"'),
+            'spark-submit foo --bar baz --password="******"',
+        ),
+        (
+            ("spark-submit", "foo", "--bar", "baz", "--password=secret"),
+            "spark-submit foo --bar baz --password=******",
+        ),
+        (
+            ("spark-submit", "foo", "--bar", "baz", "--password 'secret'"),
+            "spark-submit foo --bar baz --password '******'",
+        ),
+        (
+            ("spark-submit", "foo", "--bar", "baz", "--password secret"),
+            "spark-submit foo --bar baz --password ******",
+        ),
+        (
+            ("spark-submit", "foo", "--bar", "baz", '--password "secret"'),
+            'spark-submit foo --bar baz --password "******"',
+        ),
+        (
+            ("spark-submit", "foo", "--bar", "baz", "--secret='secret'"),
+            "spark-submit foo --bar baz --secret='******'",
+        ),
+        (
+            ("spark-submit", "foo", "--bar", "baz", "--foo.password='secret'"),
+            "spark-submit foo --bar baz --foo.password='******'",
+        ),
+        (
+            ("spark-submit",),
+            "spark-submit",
+        ),
+
+        (
+            ("spark-submit", "foo", "--bar", "baz", "--password \"secret'"),
+            "spark-submit foo --bar baz --password \"secret'",
+        ),
+        (
+            ("spark-submit", "foo", "--bar", "baz", "--password 'secret\""),
+            "spark-submit foo --bar baz --password 'secret\"",
+        ),
+    ),
+)
+def test_masks_passwords(command: str, expected: str) -> None:
```

Comment:
I figured it would be a good place to use `pytest.mark.parameterize` - [but it doesn't go along with `unittest.TestCase`](https://docs.pytest.org/en/stable/unittest.html#pytest-features-in-unittest-testcase-subclasses).

On the other hand, I get it that going with a separate function outside of the `TestSparkSubmitHook` class looks kind of awkward. Which way do we want this test to go?

Reply:
You can use https://pypi.org/project/parameterized/ in future we will probably migrate to pytest

Reply:
Done. :)

Btw when changing the regular expression, I considered these things happening in the string under masking to possibly be impacted by bugs in said expression:
* with and without the `=`
* with single quoted/double quoted/not quoted secret value
* with and without additional non-matching quotes in the middle of the value
* with and without flags/arguments after the secret value

I have added all of the combinations for now but that's quite a few - it may be better to get rid of some of them. What do you think?

Reply:
I would say that we are ok with single test case for each possibility without testing all combinations

Reply:
Sure. I tried to leave the most representative subset, does it look reasonable?